supabase/migrations/20250307021828_create_temporal_transfers_table.sql

```diff
@@ -0,0 +1,391 @@
+SET check_function_bodies = OFF;
+
+-- Create temporal schema
+CREATE SCHEMA IF NOT EXISTS temporal;
+
+-- Grant permissions for temporal schema
+GRANT USAGE ON SCHEMA temporal TO authenticated;
+GRANT USAGE ON SCHEMA temporal TO service_role;
+
+-- Grant execute on functions to service_role
+GRANT EXECUTE ON ALL FUNCTIONS IN SCHEMA temporal TO service_role;
+ALTER DEFAULT PRIVILEGES IN SCHEMA temporal
+    GRANT EXECUTE ON FUNCTIONS TO service_role;
+
+CREATE TYPE temporal.transfer_status AS ENUM(
+    'initialized',
+    'sent',
+    'confirmed',
+    'failed',
+    'cancelled'
+);
+
+CREATE TABLE temporal.send_account_transfers(
+    id serial primary key,
+    workflow_id text not null,
+    user_id uuid not null,
+    status temporal.transfer_status not null,
+    data jsonb not null,
+    created_at_block_num numeric not null,
+    created_at timestamp with time zone not null default (now() AT TIME ZONE 'utc'::text),
+    updated_at timestamp with time zone not null default (now() AT TIME ZONE 'utc'::text)
+);
+
+GRANT ALL ON TABLE temporal.send_account_transfers TO service_role;
+
+alter table "temporal"."send_account_transfers"
+    enable row level security;
+
+create policy "users can see their own temporal transfers"
+on "temporal"."send_account_transfers" as permissive
+for select to authenticated
+using (
+    auth.uid() = user_id
```

Comment:
```suggestion
    (select auth.uid()) = user_id
```

Otherwise, auth.uid will be executed for every single row in the table.